tcs-service/src/main/java/com/transformuk/hee/tis/tcs/service/api/validation/ProgrammeMembershipValidator.java

```diff
@@ -108,6 +112,25 @@ private List<FieldError> checkProgramme(ProgrammeMembershipDTO programmeMembersh
     return fieldErrors;
   }
 
+  /**
+   * Check programme start date is before finish.
+   *
+   * @param programmeMembershipDto return
+   */
+  private void checkProgrammeDates(List<FieldError> fieldErrors,
+                                   ProgrammeMembershipDTO programmeMembershipDto) {
+
+    LocalDate startDate = programmeMembershipDto.getProgrammeStartDate();
+    LocalDate endDate = programmeMembershipDto.getProgrammeEndDate();
+
+    if (startDate.isAfter(endDate)) {
```

Comment:
Are there any null checks before this / the various combinations of input that users could supply?  Just looks like an untested opportunity for an NPE.

Reply:
The DTO has @NotNull annotations on those fields, so I don't _think_ they can ever be null?

Reply:
They are required fields so they shouldn't ever be null I don't think, I can add one in for extra peace of mind.